src/runtime/runtime.ts

```diff
@@ -894,7 +818,7 @@ class JoinNode implements Node {
     return affectedConstraints;
   }
 
-  applyCombination(input:Change, prefix:ID[], transaction:number, round:number, results:ID[][]) {
+  applyCombination(index:Index, input:Change, prefix:ID[], transaction:number, round:number, results:ID[][]) {
```

Comment:
it passes it on to both presolveCheck and genericJoin. The only reason this worked before was because we had a global named "index" :(